File: api/app/schemas/ai.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

class AiChatRequest(BaseModel):
  message: str = Field(min_length=1, max_length=4000)


class AiChatResponse(BaseModel):
  reply: str
  model: str
  user_message_id: UUID
  assistant_message_id: UUID


class AiMessageOut(BaseModel):
  id: UUID
  role: str
  content: str
  created_at: datetime
  model_config = ConfigDict(from_attributes=True)

  
class AiMessagesListOut(BaseModel):
  items: list[AiMessageOut]
  total: int



class AiHealthResponse(BaseModel):
  enabled: bool
  available: bool
  model: str

File: web/app/types/ai.ts
export type AiMessageRole = 'user' | 'assistant';

export type AiMessage = {
  id: string
  role: AiMessageRole
  body: string
  created_at: string
};

export type AiMessageOut = {
  id: string
  role: string
  content: string
  created_at: string
};

export type AiMessagesListOut = {
  items: AiMessageOut[]
  total: number
};

export type AiChatResponse = {
  reply: string
  model: string
  user_message_id: string
  assistant_message_id: string
};

export type AiHealthResponse = {
  enabled: boolean
  available: boolean
  model: string
};